packages/perseus/src/widgets/matrix/matrix-validator.test.ts

```diff
@@ -0,0 +1,184 @@
+        const result = matrixValidator(userInput, rubric, mockStrings);
+
+        // Assert
+        expect(result).toHaveBeenAnsweredIncorrectly();
+    });
+
+    it("is invalid when there's an empty cell: null", () => {
+        // Arrange
+        const userInput: PerseusMatrixUserInput = {
+            answers: [
+                // TODO: this is either legacy logic or an incorrect type,
+                // but this is what the validator is checking for
+                // @ts-expect-error - TS(2322) - Type 'null' is not assignable to type 'number'.
+                [0, 0, null],
+                [0, 0, 0],
+                [0, 0, 0],
+            ],
+        };
+
+        const rubric: PerseusMatrixRubric = {
+            prefix: "",
+            suffix: "",
+            answers: [
+                [0, 1, 2],
+                [3, 4, 5],
+                [6, 7, 8],
+            ],
+            cursorPosition: [],
+            matrixBoardSize: [],
+            static: false,
+        };
+
+        // Act
+        const result = matrixValidator(userInput, rubric, mockStrings);
+
+        // Assert
+        expect(result).toHaveInvalidInput();
+    });
+
+    it("is invalid when there's an empty cell: empty string", () => {
+        // Arrange
+        const userInput: PerseusMatrixUserInput = {
+            answers: [
+                // TODO: this is either legacy logic or an incorrect type,
+                // but this is what the validator is checking for
+                // @ts-expect-error - TS(2322) - Type 'null' is not assignable to type 'number'.
+                [0, 0, ""],
+                [0, 0, 0],
+                [0, 0, 0],
+            ],
+        };
+
+        const rubric: PerseusMatrixRubric = {
+            prefix: "",
+            suffix: "",
+            answers: [
+                [0, 1, 2],
+                [3, 4, 5],
+                [6, 7, 8],
+            ],
+            cursorPosition: [],
+            matrixBoardSize: [],
+            static: false,
+        };
+
+        // Act
+        const result = matrixValidator(userInput, rubric, mockStrings);
+
+        // Assert
+        expect(result).toHaveInvalidInput();
+    });
+
+    it("is considered incorrect when the size is wrong", () => {
+        // Arrange
+        const correctUserInput: PerseusMatrixUserInput = {
+            answers: [
+                [0, 1, 2],
+                [3, 4, 5],
+                [6, 7, 8],
+            ],
+        };
+
+        const incorrectUserInput: PerseusMatrixUserInput = {
+            // Base the incorrect answer off of the correct answer.
+            // This is so we can check that it's considered incorrect
+            // if it has the wrong length, even though it otherwise
+            // would be a partial match.
+            answers: [...correctUserInput.answers, [8, 6, 7]],
+        };
+
+        const rubric: PerseusMatrixRubric = {
+            prefix: "",
+            suffix: "",
+            answers: [
```

Comment:
Nit, instead of writing out the correct answer, you could just pass in `correctUserInput.answers` since that const is meant to represent the valid answer.